src/sqlfluff/rules/L046.py

```diff
@@ -54,30 +54,32 @@ def _get_whitespace_ends(s: str) -> Tuple[str, str, str]:
         pos = main.find(inner)
         return main[:pos], inner, main[pos + len(inner) :]
 
-    def _eval(self, context: RuleContext) -> LintResult:
+    def _eval(self, context: RuleContext) -> EvalResultType:
         """Look for non-literal segments."""
         assert context.segment.pos_marker
         if context.segment.is_raw() and not context.segment.pos_marker.is_literal():
+            if not context.memory:
+                memory = set()
+            else:
```

Comment:
I know you didn't change this, and just copied it from below, but I'm curious why the other rules using `memory` do not have to initialise it like this?

Reply:
IIRC, memory defaults to an empty **dictionary**, so I'm guessing it's because this rule uses a **set** instead.